ai2thor_wrapper/envs.py

```diff
@@ -0,0 +1,195 @@
+import os
+import json
+import configparser
+import sys
+
+import numpy as np
+import skimage.color, skimage.transform
+import ai2thor.controller
+
+POSSIBLE_ACTIONS = [
+    'MoveAhead',
+    'MoveBack',
+    'MoveRight',
+    'MoveLeft',
+    'LookUp',
+    'LookDown',
+    'RotateRight',
+    'RotateLeft',
+    'OpenObject',
+    'CloseObject',
+    'PickupObject',
+    'PutObject'
+    # Teleport and TeleportFull but these shouldn't be actions for an agent
+]
+
+class ThorWrapperEnv():
+    def __init__(self, scene_id='FloorPlan28', seed=None, task=0, max_episode_length=1000, current_object_type='Mug',
+                 grayscale=True, interaction=True, config_path='config_example.ini', movement_reward=0):
+        # Loads config file from path relative to ai2thor_wrapper python package folder
+        config_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), config_path)
+        self.config = configparser.ConfigParser()
+        config_output = self.config.read(config_path)
+        if len(config_output) == 0:
+            print('No config file found at: {}. Exiting'.format(config_path))
+            sys.exit()
+
+        self.scene_id = scene_id
+        self.controller = ai2thor.controller.Controller()
+        self.controller.start()
+        if seed:
+            self.seed(seed)
+
+        self.controller.reset(self.scene_id)
+        self.event = self.controller.step(dict(action='Initialize', gridSize=0.25, renderDepthImage=True,
+                                               renderClassImage=True,
+                                               renderObjectImage=True))
+
+        self.current_object_type = current_object_type
+        self.max_episode_length = max_episode_length
+        self.t = 0
+        self.done = False
+        self.task = task
+        self.movement_reward = movement_reward
+
+        # action dictionary from int to dict action to pass to event.controller.step()
+        self.ACTION_SPACE = {}
+        for idx, action_str in enumerate(POSSIBLE_ACTIONS):
+            self.ACTION_SPACE[idx] = dict(action=action_str)
+
+        if not interaction:
```

Comment:
I would solve the interaction thing differently. Create two lists: moving_actions and interaction_actions. Then:
```suggestion
        possible_actions = moving_actions.copy()
        if interaction:
            possible_actions.extend(interaction_actions)
```
And continue with the dict comprehension as before

Reply:
seems a bit more complex rather than just "removing interaction actions if no interaction allowed" no?